plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -517,6 +517,7 @@ private IcebergSplit toIcebergSplit(FileScanTask task, TupleDomain<IcebergColumn
                 SplitWeight.fromProportion(clamp((double) task.length() / tableScan.targetSplitSize(), minimumAssignedSplitWeight, 1.0)),
                 fileStatisticsDomain,
                 fileIoProperties,
-                cachingHostAddressProvider.getHosts(task.file().path().toString(), ImmutableList.of()));
+                cachingHostAddressProvider.getHosts(task.file().path().toString(), ImmutableList.of()),
+                task.file().dataSequenceNumber());
```

Comment:
Here we are casting a `Long` (see `org.apache.iceberg.ContentFile#dataSequenceNumber`) to `long` which can potentially cause a NPE